core/sql/optimizer/BindRelExpr.cpp

```diff
@@ -13445,6 +13435,14 @@ RelExpr * GenericUpdate::bindNode(BindWA *bindWA)
   if (getOperatorType() == REL_UNARY_UPDATE ||
       getOperatorType() == REL_UNARY_DELETE) {
 
+    if (getTableDesc()->getNATable()->isHiveTable())
+      {
+        *CmpCommon::diags() << DgSqlCode(-4223)
+                            << DgString0("Update/Delete on Hive table is");
+        bindWA->setErrStatus();
+        return this;
+      }
+
```

Comment:
During binder phase, relational nodes are typed as REL_UNARY_UPDATE or REL_UNARY_DELETE
for update or delete operations. Binder checks for them to determine if it is an update or delete.
They may later get changed to other upd/del like subset/unique/cursor etc.
No update or delete operations are currently supported on hive tables.